managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -2667,8 +2667,10 @@ void internalTrimLedgers(boolean isTruncate, CompletableFuture<?> promise) {
                     ledgers.headMap(slowestReaderLedgerId, false).values().iterator();
             while (ledgerInfoIterator.hasNext()){
                 LedgerInfo ls = ledgerInfoIterator.next();
-                // currentLedger can not be deleted
-                if (ls.getLedgerId() == currentLedger.getId()) {
+                // Current ledger can not be deleted when it is currently being written to.
+                // However, When the manager ledger state is ClosedLedger, the current ledger was closed
+                // and there are no pending operations. So it can be deleted.
+                if (ls.getLedgerId() == currentLedger.getId() && currentState != State.ClosedLedger) {
```

Comment:
The managed ledger's `current ledger` is still pointed at this ledger, but you deleted it. 

You can trigger a Current ledger switch if you just want to free the storage usage.

Reply:
I have considered this when writing this question, but what I am thinking is whether it is necessary to create a new ledegr if there are no new messages. Will creating a new ledger add unnecessary expenses.

Reply:
But for normal production and consumption scenarios, there should be no situation where the new ledger is not written to for a long time. Users can handle this situation even for inactive topics according to the inactive_topic_policies.